File: frontend/src/raid/ColorMatchSystem.js
import { isGameColor } from '../colors.js';

/**
 * Exact camouflage match. Unpainted tiles (null) never match.
 * Independent of rendering — always uses stored gameplay colors.
 */
export function isMatch(attackerColor, tileColor) {
  if (!isGameColor(attackerColor)) return false;
  if (tileColor == null || tileColor === '') return false;
  return String(attackerColor).trim().toUpperCase() === String(tileColor).trim().toUpperCase();
}

export const ColorMatchSystem = { isMatch };
export default ColorMatchSystem;

File: backend/src/main/java/com/shadowpalette/stealth/ColorMatchSystem.java
package com.shadowpalette.stealth;

import com.shadowpalette.util.Colors;

public final class ColorMatchSystem {

    private ColorMatchSystem() {}

    public static boolean isMatch(String attackerColor, String tileColor) {
        if (attackerColor == null || tileColor == null || tileColor.isBlank()) {
            return false;
        }
        String a = attackerColor.trim().toUpperCase();
        String t = tileColor.trim().toUpperCase();
        if (!Colors.ALLOWED_CAMO_COLORS.contains(a)) {
            return false;
        }
        return a.equals(t);
    }
}
